### PR TITLE
[NT- 939] Creator details tracking events

### PR DESCRIPTION
**Pull request by cdolm92**
# 📲 What

Adds`Creator Details Clicked` to be tracked via Optimizely and hides `creatorButton` if the user is in the experiment variant.

# 🤔 Why

In order to track the success of our Project page experiment.

# 🛠 How

Adds this event and its respective attributes and event tags. As well as tests.

# ✅ Acceptance criteria

- [ ]  Confirm that the attributes are correct
- [ ]  Confirm that these events are being tracked to Optimizely
- [ ]  Confirm that the `creatorButton` is shown when user is in  the`control` variant
- [ ]  Confirm that the  `creatorButton` is hidden when user is in the`variation-1` variant

### CODE REVIEW

File: Kickstarter-iOS/Views/Cells/ProjectPamphletMainCell.swift [hunk 1]
```diff
@@ -34,6 +34,10 @@ internal final class ProjectPamphletMainCell: UITableViewCell, ValueCell {
       |> \.translatesAutoresizingMaskIntoConstraints .~ false
   }()
 
+  private lazy var tapGesture: UITapGestureRecognizer = {
```
Comment (justinswart) [inline Kickstarter-iOS/Views/Cells/ProjectPamphletMainCell.swift]: Could we call this `creatorBylineTapGesture` just to make it easier to know what this is in future.

**Comment by nativeksr** [comment]
### SwiftFormat found issues:

| File | Rules |
| ---- | ----- |
| Library/ViewModels/ProjectPamphletMainCellViewModel.swift | blankLinesBetweenScopes |
| Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift | indent |


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/ViewModels/ProjectPamphletMainCellViewModel.swift [hunk 2]
```diff
@@ -47,6 +50,9 @@ public protocol ProjectPamphletMainCellViewModelOutputs {
   /// Emits a string for the conversion label.
   var conversionLabelText: Signal<String, Never> { get }
 
+  /// Emits a bool to determine that creatorButton should be hidden.
+  var creatorButtonIsHidden: Signal<Bool, Never> { get }
```
Comment (justinswart) [inline Library/ViewModels/ProjectPamphletMainCellViewModel.swift]: Not sure if we need this? Isn't the button a subview of `creatorStackView` which will be hidden?

Comment (cdolm92) [reply Library/ViewModels/ProjectPamphletMainCellViewModel.swift]: It's not 
Screen Shot 2020-03-06 at 2 49 22 PM

File: Library/ViewModels/ProjectPamphletMainCellViewModel.swift [hunk 3]
```diff
@@ -161,6 +170,10 @@ public final class ProjectPamphletMainCellViewModel: ProjectPamphletMainCellView
         creatorDetails == nil || isLoading
       }
 
+    self.creatorButtonIsHidden = projectAndCreatorDetails.map { _, creatorDetails in
```
Comment (justinswart) [inline Library/ViewModels/ProjectPamphletMainCellViewModel.swift]: Could we just make this:

```swift
self.creatorButtonIsHidden = self.creatorStackViewHidden
```
That way we know the exact same behaviour of hiding/showing the stackview is applied to the button.

File: Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift [hunk 4]
```diff
@@ -708,6 +770,34 @@ final class ProjectPamphletMainCellViewModelTests: TestCase {
...
+    let optimizelyClient = MockOptimizelyClient()
+      |> \.experiments .~ [
+        OptimizelyExperiment.Key.nativeProjectPageConversionCreatorDetails.rawValue: OptimizelyExperiment.Variant.control.rawValue
+      ]
+
+    withEnvironment(optimizelyClient: optimizelyClient) {
+      self.vm.inputs.configureWith(value: (.template, nil, (nil, true)))
+      self.vm.inputs.awakeFromNib()
+
+      self.creatorButtonIsHidden.assertValues([false])
```
Comment (justinswart) [inline Library/ViewModels/ProjectPamphletMainCellViewModelTests.swift]: We can move these tests into the `testCreatorDetailsVisibility_NilCreatorDetails` and `testCreatorDetailsVisibility_NonNilCreatorDetails` tests below to just test all the UI visibility together.

**Pull request merged**